core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -446,19 +446,20 @@ public void close() {
   }
 
   /**
-   * Periodic log config check report.
+   * Periodically log the config check report.
    */
   private final class LogConfigReportHeartbeatExecutor implements HeartbeatExecutor {
-
-    /**
-     * Constructs a new {@link LogConfigReportHeartbeatExecutor}.
-     */
-    public LogConfigReportHeartbeatExecutor() {
-    }
+    private boolean mFirst = true;
```

Comment:
Does this need to be volatile?

Reply:
I don't think so since it is used from one thread only. I've updated to volatile though in case we change the way we run heartbeats in the future